### DISCUSSION TITLE
Modelling a mixed integer nonconvex problem

### DISCUSSION THREAD

**Discussion by Mark531** [cvxpy/cvxpy#1905] [Q&A] [answered]
Hello,

The following optimization problem generates a DCP error because the objective function is not convex:

```
import pyscipopt
import numpy as np
import cvxpy as cp

x = np.random.uniform(0, 1, 100)
y = 5 * x**2 - 3 * x + 4
v = cp.Variable(integer=True)
w = cp.Variable(1, boolean=True)
problem = cp.Problem(cp.Minimize(cp.norm(y - v) * w))
problem.solve(solver=cp.SCIP)
```

Of course, this problem makes no sense, it just shows what I'm trying to do: activating/deactivating optimization on some parts of the objective function with a boolean variable. Since v and w and not directly multiplied together, I can't create a third variable modelling the product. Is it possible to rewrite the problem or is it impossible to solve with the current available solvers?

**Comment by SteveDiamond** [comment, marked as answer]
You can definitely make it work. For example, you could write ``cp.pos(cp.norm(y - v) - M * w)`` where ``M`` is very large. I'm sure there are better ways as well. This resource might be helpful https://docs.mosek.com/modeling-cookbook/mio.html
